src/Sarif.Multitool/PageCommand.cs

```diff
@@ -0,0 +1,195 @@
+
+            // Write the desired page from the Sarif file
+            ExtractPage(options, root);
+        }
+
+        internal SarifLog PageViaOm(PageOptions options)
+        {
+            SarifLog actualLog = ReadSarifFile<SarifLog>(_fileSystem, options.InputFilePath);
+
+            // Validate Run in range
+            if (options.RunIndex >= actualLog?.Runs.Count)
+            {
+                throw new ArgumentOutOfRangeException($"Page requested for RunIndex {options.RunIndex}, but Log had only {actualLog.Runs?.Count} runs.");
+            }
+
+            // Filter to desired run only
+            Run run = actualLog.Runs[options.RunIndex];
+            actualLog.Runs = new List<Run>() { run };
+
+            // Validate results in range
+            if (options.Index + options.Count > run?.Results.Count)
+            {
+                throw new ArgumentOutOfRangeException($"Page requested from Result {options.Index} to {options.Index + options.Count}, but Run has only {run.Results?.Count} results.");
+            }
+
+            // Filter to desired results only
+            run.Results = run.Results.Skip(options.Index).Take(options.Count).ToList();
+
+            WriteSarifFile(_fileSystem, actualLog, options.OutputFilePath, Formatting.None);
+            return actualLog;
+        }
+
+        private JsonMapNode LoadOrRebuildMap(PageOptions options, string mapPath)
+        {
+            JsonMapNode root;
+            Stopwatch w = Stopwatch.StartNew();
+
+            if (_fileSystem.FileExists(mapPath) && _fileSystem.GetLastWriteTime(mapPath) > _fileSystem.GetLastWriteTime(options.InputFilePath))
+            {
+                // If map exists and is up-to-date, just reload it
+                Console.WriteLine($"Loading Json Map \"{mapPath}\"...");
+                root = JsonConvert.DeserializeObject<JsonMapNode>(_fileSystem.ReadAllText(mapPath));
+            }
+            else
+            {
+                // Otherwise, build the map and save it
+                Console.WriteLine($"Building Json Map of \"{options.InputFilePath}\" into \"{mapPath}\"...");
+                JsonMapBuilder builder = new JsonMapBuilder(options.TargetMapSizeRatio);
+                root = builder.Build(() => _fileSystem.OpenRead(options.InputFilePath));
+
+                if (root != null)
+                {
+                    _fileSystem.WriteAllText(mapPath, JsonConvert.SerializeObject(root, Formatting.None));
+                }
+            }
+
+            w.Stop();
+            Console.WriteLine($"Done in {w.Elapsed.TotalSeconds:n1}s.");
+
+            return root;
+        }
+
+        private void ExtractPage(PageOptions options, JsonMapNode root)
+        {
+            Stopwatch w = Stopwatch.StartNew();
+            Console.WriteLine($"Extracting Page [{options.Index}, {options.Index + options.Count}) from \"{options.InputFilePath}\" into \"{options.OutputFilePath}\"...");
```

Comment:
>) [](start = 97, length = 1)

this doesn't match with the open square bracket " [ ",

Reply:
This was formal mathematical notation. I'm writing and naming the arguments separately now to be clearer:
$"Extracting {options.Count:n0} results from index {options.Index:n0}\r\n  from \"{options.InputFilePath}\"\r\n  into \"{options.OutputFilePath}\"..."